pkg/helper/log/task_reader.go

```diff
@@ -41,87 +42,70 @@ func (s *step) hasStarted() bool {
+	tkn := r.clients.Tekton
+	tr, err := tkn.TektonV1alpha1().TaskRuns(r.ns).Get(r.run, metav1.GetOptions{})
 	if err != nil {
-		return nil, nil, fmt.Errorf("%s: %s", msgTRNotFoundErr, err)
+		return nil, nil, fmt.Errorf("%s: %s", MsgTRNotFoundErr, err)
 	}
 
-	lr.formTaskName(tr)
-
-	return lr.readLogs(tr)
-}
+	r.formTaskName(tr)
 
-func (lr *LogReader) readLogs(tr *v1alpha1.TaskRun) (<-chan log.Log, <-chan error, error) {
-	if lr.Follow {
-		return lr.readLiveLogs()
+	if r.follow {
+		return r.readLiveTaskLogs()
 	}
-	return lr.readAvailableLogs(tr)
+	return r.readAvailableTaskLogs(tr)
 }
 
-func (lr *LogReader) formTaskName(tr *v1alpha1.TaskRun) {
-	if lr.Task != "" {
+func (r *Reader) formTaskName(tr *v1alpha1.TaskRun) {
+	if r.task != "" {
 		return
 	}
 
 	if name, ok := tr.Labels["tekton.dev/pipelineTask"]; ok {
-		lr.Task = name
+		r.task = name
 		return
 	}
 
 	if tr.Spec.TaskRef != nil {
-		lr.Task = tr.Spec.TaskRef.Name
+		r.task = tr.Spec.TaskRef.Name
 		return
 	}
 
-	lr.Task = fmt.Sprintf("Task %d", lr.Number)
+	r.task = fmt.Sprintf("Task %d", r.number)
 }
 
-func (lr *LogReader) readLiveLogs() (<-chan log.Log, <-chan error, error) {
-	tr, err := lr.waitUntilPodNameAvailable(10)
+func (r *Reader) readLiveTaskLogs() (<-chan Log, <-chan error, error) {
+	tr, err := r.waitUntilTaskPodNameAvailable(10)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	var (
 		podName = tr.Status.PodName
-		kube    = lr.Clients.Kube
+		kube    = r.clients.Kube
 	)
 
-	p := pods.New(podName, lr.Ns, kube, lr.Streamer)
+	p := pods.New(podName, r.ns, kube, r.streamer)
 	pod, err := p.Wait()
 	if err != nil {
-		return nil, nil, errors.New(fmt.Sprintf("task %s failed: %s. Run tkn tr desc %s for more details.", lr.Task, strings.TrimSpace(err.Error()), tr.Name))
+		return nil, nil, errors.New(fmt.Sprintf("task %s failed: %s. Run tkn tr desc %s for more details.", r.task, strings.TrimSpace(err.Error()), tr.Name))
 	}
 
-	steps := filterSteps(pod, lr.AllSteps, lr.Steps)
-	logC, errC := lr.readStepsLogs(steps, p, lr.Follow)
+	steps := filterSteps(pod, r.allSteps, r.steps)
+	logC, errC := r.readStepsLogs(steps, p, r.follow)
 	return logC, errC, err
 }
 
-func (lr *LogReader) readAvailableLogs(tr *v1alpha1.TaskRun) (<-chan log.Log, <-chan error, error) {
+func (r *Reader) readAvailableTaskLogs(tr *v1alpha1.TaskRun) (<-chan Log, <-chan error, error) {
 	if !tr.HasStarted() {
-		return nil, nil, fmt.Errorf("task %s has not started yet", lr.Task)
+		return nil, nil, fmt.Errorf("task %s has not started yet", r.task)
 	}
 
-	//Check if taskrun failed on start up
-	if err := hasTaskRunFailed(tr.Status.Conditions, lr.Task); err != nil {
-		if lr.Stream != nil {
-			fmt.Fprintf(lr.Stream.Err, "%s\n", err.Error())
+	// Check if taskrun failed on start up
+	if err := hasTaskRunFailed(tr.Status.Conditions, r.task); err != nil {
+		if r.stream != nil {
+			fmt.Fprintf(r.stream.Err, "%s\n", err.Error())
 		}
 		return nil, nil, err
```

Comment:
Isn't ```failed``` pod log shown again without ```-f``` flag 🤔 ?
(because return at this point before readStepsLogs)

following code is correct 🤔 ?
```
if r.stream != nil {
	fmt.Fprintf(r.stream.Err, "%s\n", err.Error())
} else {
	return nil, nil, err
}
```

Reply:
Yes, that was something that was overlooked on my part in #721. I think adding the else block makes sense. 

Overall, we need to clean up error messaging in logs/have better testing around this. Maybe even an e2e scenario.

Reply:
> Yes, that was something that was overlooked on my part in #721. I think adding the else block makes sense.

Oh, Thanks !!
I'll add else block in this PR for now :thumbsup:

> Overall, we need to clean up error messaging in logs/have better testing around this. Maybe even an e2e scenario.

+1